File: neoforge/src/main/java/wile/redstonepen/platform/PlatformHelperNeoForge.java
package wile.redstonepen.platform;

import wile.redstonepen.libmc.*;

import net.minecraft.server.level.ServerLevel;
import net.minecraft.world.entity.player.Player;
import net.neoforged.fml.loading.FMLLoader;
import net.neoforged.neoforge.common.util.FakePlayerFactory;
import net.neoforged.fml.ModList;

import java.nio.file.Path;
import java.util.Optional;

public class PlatformHelperNeoForge implements IPlatformHelper
{
  @Override
  public Path getGameDirectory()
  { return FMLLoader.getGamePath(); }

  @Override
  public boolean isModLoaded(String modId)
  { return ModList.get().isLoaded(modId); }

  @Override
  public Optional<? extends Player> getFakePlayer(ServerLevel world)
  {
    try { return Optional.of(FakePlayerFactory.getMinecraft(world)); }
    catch(Exception e) { return Optional.empty(); }
  }
}

File: neoforge/src/main/kotlin/wile/redstonepen/neoforge/PlatformHelperNeoForge.kt
package wile.redstonepen.neoforge

import java.nio.file.Path
import java.util.Optional
import net.minecraft.server.level.ServerLevel
import net.minecraft.world.entity.player.Player
import net.neoforged.fml.ModList
import net.neoforged.fml.loading.FMLLoader
import net.neoforged.neoforge.common.util.FakePlayerFactory
import wile.redstonepen.platform.IPlatformHelper

class PlatformHelperNeoForge : IPlatformHelper {
    override fun getGameDirectory(): Path = FMLLoader.getGamePath()

    override fun isModLoaded(modId: String): Boolean = ModList.get().isLoaded(modId)

    override fun getFakePlayer(world: ServerLevel): Optional<out Player> =
        try {
            Optional.of(FakePlayerFactory.getMinecraft(world))
        } catch (ignored: Exception) {
            Optional.empty()
        }
}
